packages/mirror-media-next/utils/login.js

```diff
@@ -0,0 +1,25 @@
+export function detectMobileOs() {
+  const userAgent = navigator.userAgent
+
+  if (/iPad|iPhone|iPod/.test(userAgent)) {
+    return 'ios'
+  } else if (/android/i.test(userAgent)) {
+    // Redirect to the Google Play Store
+    return 'android'
+  } else {
+    return 'non-mobile'
+  }
+}
+
+export function isInWebView() {
+  const userAgent = window.navigator.userAgent || ''
```

Comment:
需要附上參考資料來源，如這邊所示
https://github.com/mirror-media/mirror-media-nuxt/blob/5a1b4d0b832260064670b83f353d6c251993837b/plugins/user-behavior-log/util/is-in-app-browser.js#L1

Reply:
實作方式是從 chatGPT 取得，因為網上看到的做法都不盡相同，因此想說直接測試可行的方式最準，不過既然 mm-nuxt曾經有實作過一樣的邏輯就直接改成這個做法就好，感謝建議，改動在 [a015298](https://github.com/mirror-media/Adam/pull/1077/commits/a01529843950e384655dcadbc05618a82381fdcd)

Reply:
我會建議用上這兩者的資訊，讓 user agent 的偵測覆蓋範圍更廣。
https://gist.github.com/fostyfost/0591c79f4cd7ca26e5941a53fd4bf1a4
https://gist.github.com/monkianer/c163651c86897d744f10

Reply:
想確認一下你覺得要放入的檢查機制是兩者全部的條件的聯集嗎，因為看起來前者跟原本的 mm-nuxt 寫得差不多，只是多了很舊版本的支援 (我們有需要支援到這個舊的版本嗎？)，後者目前看下來應該只有 FB, line 幾個比較有需要加入，但也很懷疑是不是有需要另外加入，說不定前者的 rule 其實就有包含了這些邏輯，想確認一下你的想法？

前者
```
const rules = [
  // If it says it's a webview, let's go with that.
  'WebView',
  // iOS webview will be the same as safari but missing "Safari".
  '(iPhone|iPod|iPad)(?!.*Safari)',
  // https://developer.chrome.com/docs/multidevice/user-agent/#webview_user_agent
  'Android.*Version/[0-9].[0-9]',
  // Also, we should save the wv detected for Lollipop.
  // Android Lollipop and Above: webview will be the same as native,
  // but it will contain "wv".
  'Android.*wv',
  // Old chrome android webview agent
  'Linux; U; Android',
]
```

後者
```
    return { //偵測移動端瀏覽器版本信息
        trident: u.indexOf('Trident') > -1, //IE 核心
        presto: u.indexOf('Presto') > -1, //opera 核心
        webKit: u.indexOf('AppleWebKit') > -1, //Apple, google 核心
        gecko: u.indexOf('Gecko') > -1 && u.indexOf('KHTML') == -1, //Firefox 核心
        mobile: !!u.match(/AppleWebKit.*Mobile.*/), //行動裝置
        ios: !!u.match(/\(i[^;]+;( U;)? CPU.+Mac OS X/), //ios
        android: u.indexOf('Android') > -1 || u.indexOf('Linux') > -1, //android或uc瀏覽器
        iPhone: u.indexOf('iPhone') > -1, //是否為iPhone或者QQHD瀏覽器
        iPad: u.indexOf('iPad') > -1, //是否iPad
        webApp: u.indexOf('Safari') == -1, //是否web應用程式
        iosv: u.substr(u.indexOf('iPhone OS') + 9, 3),//ios版本
        weixin: ua.match(/MicroMessenger/i) == "micromessenger",//微信瀏覽器
        fbapp: u.indexOf('FBAV') > -1,//Facebook App內瀏覽器
        line: u.indexOf('Line') > -1//Line內瀏覽器
    };
```

Reply:
前者全部加上後者的 fbapp 和 line 應該就可以了

Reply:
修改在 [36f245f](https://github.com/mirror-media/Adam/pull/1077/commits/36f245fff39096fe66f5993909be5f41e54c0e23)